Update source setup page in lisk docs (#1576)
LiskHQ/lisk-service: docs/antora/modules/ROOT/pages/setup/source.adoc

====

[tabs]
====
Ubuntu::
+
--
[source,bash]
----
sudo apt update
sudo apt install mysql-server=8.0*
sudo mysql_secure_installation
----

--
macOS::
+
--
[source,bash]
----
brew tap homebrew/services
brew install mysql@8.0
brew services start mysql@8.0
----

.Authentication

If you encounter issues to authenticate, and received the following error:

 caching_sha2_password' cannot be loaded: dlopen(/usr/local/lib/plugin/caching_sha2_password.so, 2): image not found

Try the following: Change the `default_authentication_plugin` using the `mysql_native_password`.

Open up `my.cnf` .

If you are unsure where to find your `my.cnf`, use the following command:

 $ mysql --verbose --help | grep my.cnf

Add the following at the end of the file:

 default_authentication_plugin=mysql_native_password

Save and exit.

Next, login via the terminal:

 $ mysql -uroot

Then run the following command to update the root password:

 ALTER USER 'root'@'localhost' IDENTIFIED WITH mysql_native_password BY '';

Now you should be able to login to your MySQL 8 via your MySQL Client.

> 

Reviewer: Is this only relevant for macOS users?

Author: Good point, it is actually relevant for all platforms, so I have updated line 187 to reflect this 👍

> .Authentication
> 
> If you encounter issues authenticating, regardless of which platform you are using, and you received the following error:
> 
>  caching_sha2_password' cannot be loaded: dlopen(/usr/local/lib/plugin/caching_sha2_password.so, 2): image not found
> 
> Try to perform the following: Change the `default_authentication_plugin` using the `mysql_native_password`.
> 
> Open up `my.cnf` .
> 
> If you are unsure where to find your `my.cnf`, run the following command:
> 
>  $ mysql --verbose --help | grep my.cnf
> 
> Add the following at the end of the file:
> 
>  default_authentication_plugin=mysql_native_password
> 
> Save and exit.
> 
> Next, login via the terminal:
> 
>  $ mysql -uroot
> 
> Then run the following command to update the root password:
> 
>  ALTER USER 'root'@'localhost' IDENTIFIED WITH mysql_native_password BY '';
> 
> Now you should be able to login to your MySQL 8 via your MySQL Client.
> 
> 
> --
> ====

Reviewer: If this is true for both platforms, please move the statement out of the tabs, so it is visible below no matter what tab is selected :)

```suggestion
--
====

.Authentication

If you encounter issues authenticating, and you received the following error:

 caching_sha2_password' cannot be loaded: dlopen(/usr/local/lib/plugin/caching_sha2_password.so, 2): image not found

Try to perform the following: Change the `default_authentication_plugin` using the `mysql_native_password`.

Open up `my.cnf` .

If you are unsure where to find your `my.cnf`, run the following command:

 $ mysql --verbose --help | grep my.cnf

Add the following at the end of the file:

 default_authentication_plugin=mysql_native_password

Save and exit.

Next, login via the terminal:

 $ mysql -uroot

Then run the following command to update the root password:

 ALTER USER 'root'@'localhost' IDENTIFIED WITH mysql_native_password BY '';

Now you should be able to login to your MySQL 8 via your MySQL Client.

```

Author: Yes, good point Mona...Now done 👍

=== GeoJS

{url_geojs}[GeoJS] is used by the Network Monitor for IP address geo-location.

[source,bash]
----
#todo
----
////

=== PM2

{url_pm2}[PM2] helps manage the node processes for Lisk Service and also offers easy log rotation (Highly Recommended).

[source,bash]
----
npm install -g pm2
----

== Installation

[IMPORTANT]
====
It is strongly recommended that you synchronize your Lisk Core node with the network **before** starting the Lisk Service.
====

If you have not already done so, clone the {url_github_service}[lisk-service^] GitHub repository and then navigate into the project folder and check out the latest release.

[source,bash]
----
# Clone Lisk Service repository
git clone https://github.com/LiskHQ/lisk-service.git

# Change directory to the new repository
cd lisk-service

# Switch to the recent stable as a base
git checkout v0.7.0

# ...or use the development branch
git checkout development
----

Install all npm dependencies from the root directory.

[source,bash]
----
make build-local
----

Now it is possible to start Lisk Service.

> .Start Lisk Service from Source code
> [source,bash]
> ----
> npm run start

Reviewer 2: ```suggestion
.Start Lisk Service from source code using pm2 from the root directory.
[source,bash]
----
pm2 start ecosystem.config.js
```

Reviewer 3: I think `npm run start` will do the same. Ref: https://github.com/LiskHQ/lisk-service/blob/feature/1157-prepare-lisk-service-v070-for-beta/package.json#L27

Reviewer 2: Then, we can leave it as is. @chris529 please ignore the suggestion.

----

> This will use the default configuration and connect Lisk Service to the Lisk Mainnet.

Reviewer 2: ```suggestion
The default configuration in `ecosystem.config.js` should suffice in most of the cases. If necessary, please modify the file to configure the necessary environment variables for each microservice as per your requirements.
```

To change the default configuration, check out the page xref:{url_config}[Configuration with PM2].

More commands about how to manage Lisk Service are described on the xref:{url_management_pm2}[PM2 commands] page.

> TIP: Check the xref:{url_index_usage}[Usage] section for examples of how to use and interact with Lisk Service.

Reviewer: Please comment out until the corresponding content is updated as well

Author: Good point Mona..Now done 👍

apt update
----
--
macOS::
+
--
Install https://brew.sh/[Brew] by following the https://brew.sh/[latest instructions].

Next, ensure your package sources are up to date:

[source,bash]
----
brew update
brew doctor
----
--
====

=== Tool chain components

[tabs]
====
Ubuntu::
+
--
Install the `build-essential` package alongside with several development tools.

[source,bash]
----
sudo apt-get install -y build-essential git make
----
--
macOS::
+
--
[source,bash]
----
xcode-select --install
----
--
====

=== Node.js

{url_nodejs}[Node.js^] serves as the underlying engine for code execution.
There are several different methods and version managers used to install Node.js on your system.
It is recommended to use one of the following two options:

[tabs]
====
Option A - Node version manager::
+
--
It is recommended to use a Node version manager such as {url_nvm}[NVM^].
NVM is a bash script that enables the management of multiple active Node.js versions.

. Install nvm following the {url_nvm_instructions}[official instructions^].
. Install v16 of Node.js using NVM:

.Check for the latest LTS version
[source,bash]
----
nvm ls-remote
----

.Install the latest LTS version
[source,bash]
----

> nvm install 16.15.0

Reviewer 2: ```suggestion
nvm install 16
```

----
--
Option B - Node.js package::
+
--
If NVM or other package managers are not required, it is possible to install the Node package globally  as shown in the following commands below:

*Ubuntu*

[source,bash]
----
curl -sL https://deb.nodesource.com/setup_16.x | sudo -E bash -
sudo apt-get install -y nodejs
----

*macOS*

For macOS, please execute the following command below:

[source,bash]
----

> brew install node@12

Reviewer 2: ```suggestion
brew install node@16
```

----
--
====

=== MYSQL
MySQL is used for storing persistent data.

[NOTE]
====

> Lisk Service expects a MySQL user `lisk` with password `password` and a database `lisk`.

Reviewer 2: ```suggestion
Lisk Service, by default, expects a MySQL user `lisk` with password `password` and a database `lisk`.
```

**The MySQL user `lisk` needs to be manually created** after installing MySQL.

> These values can be updated by setting the environment variable `SERVICE_CORE_MYSQL`, see the xref:{url_references_config}[]

Reviewer 2: ```suggestion
You can choose to create a custom database with a different user and password. These values can be updated by setting the following environment variables (specific to each microservice): `SERVICE_INDEXER_MYSQL`, `SERVICE_APP_REGISTRY_MYSQL` and `SERVICE_STATISTICS_MYSQL` with the appropriate MySQL connection string. See the xref:{url_references_config}[]
```

====

[tabs]
====
Ubuntu::
+
--
[source,bash]
----
sudo apt update

> sudo apt install mysql-server

Reviewer 2: ```suggestion
sudo apt install mysql-server=8.0*
```

Try the following: Change the `default_authentication_plugin` using `mysql_native_password`.

Open up `my.cnf` .

If you are unsure where to find your `my.cnf`, use the following command:

 $ mysql --verbose --help | grep my.cnf

Add the following at the end of the file:

 default_authentication_plugin=mysql_native_password

Save and exit.

Next, you need to login via the terminal:

 $ mysql -uroot

Then run the following command to update the root password:

 ALTER USER 'root'@'localhost' IDENTIFIED WITH mysql_native_password BY '';

Now you should be able to login to your MySQL 8 via your MySQL Client.
****

--
====

=== Redis

{url_redis}[Redis] is used for caching temporary data.

[tabs]
====
Redis with Docker::
+
--
**Docker Setup**

Follow the steps described in the xref:{url_setup_docker_docker}[Prerequisites > Docker] section of the "Installation with Docker" page.

**Installation**

.How to install and start Redis with Docker
[source,bash]
----
docker run --name redis_service --port 6379:6379 -d redis:5-alpine
----
.How to use the custom redis.conf file
[source,bash]
----

> docker run --name redis_service -v /path/to/custom_redis.conf:/usr/local/etc/redis/redis.conf -d redis:5-alpine

Reviewer 2: ```suggestion
docker run --name redis_service -v /path/to/custom_redis.conf:/usr/local/etc/redis/redis.conf --port 6379:6379 -d redis:5-alpine
```

----
The above commands should be enough to start Redis which is ready to use with Lisk Service.

To stop the Docker container again, execute the following commands below:

.How to stop Redis with Docker
[source,bash]
----
docker stop redis_service
----
--
Redis system-wide::
+
--
*Ubuntu*

[source, bash]
----
sudo apt-get install redis-server
----

*macOS*

[source, bash]
----
brew install redis
----
--
====

////
Lisk Service is not compatible with this service right now.
we should encourage community to make Lisk Service compatible with this service, then they can use it as alternative GeoIP service.
=== GeoJS

{url_geojs}[GeoJS] is used by the Network Monitor for IP address geo-location.

[source,bash]
----
#todo
----
////

=== PM2

> {url_pm2}[PM2] manages the node process for Lisk Service and handles log rotation (Highly Recommended).

Reviewer 2: ```suggestion
{url_pm2}[PM2] helps manage the node processes for Lisk Service and also offers easy log rotation (Highly Recommended).
```
